chia/server/address_manager_store.py

```diff
@@ -97,52 +78,105 @@ async def create_address_manager(cls, peers_file_path: Path) -> AddressManager:
+        await write_file_async(peers_file_path, out.getvalue(), file_mode=0o644)
 
+    @classmethod
+    async def deserialize_bytes(cls, data: io.BytesIO) -> AddressManager:
+        address_manager = AddressManager()
+
+        address_manager.key = int.from_bytes(data.read(32), byteorder="big")
+
+        address_manager.new_count = uint64.parse(data)
+        # deserialize new_table
+        new_table_count = uint32.parse(data)
+        new_table_nodes: list[tuple[uint64, uint64]] = []
+        for i in range(0, new_table_count):
+            node_id = uint64.parse(data)
+            bucket = uint64.parse(data)
+            new_table_nodes.append((node_id, bucket))
+
+        # deserialize node info
+        address_manager.id_count = 0
+        length = len(data.getvalue())
+        while data.tell() < length:
+            # breakpoint()
+            info = ExtendedPeerInfo.parse(data)
+            # check if we're a new node
+            if address_manager.id_count < address_manager.new_count:
+                address_manager.map_addr[info.peer_info.host] = address_manager.id_count
+                address_manager.map_info[address_manager.id_count] = info
+                info.random_pos = len(address_manager.random_pos)
+                address_manager.random_pos.append(address_manager.id_count)
+                address_manager.id_count += 1
+            else:
+                # we're a tried node
+                tried_bucket = info.get_tried_bucket(address_manager.key)
+                tried_bucket_pos = info.get_bucket_position(address_manager.key, False, tried_bucket)
+                if address_manager.tried_matrix[tried_bucket][tried_bucket_pos] == -1:
+                    info.random_pos = len(address_manager.random_pos)
+                    info.is_tried = True
+                    id_count = address_manager.id_count
+                    address_manager.random_pos.append(id_count)
+                    address_manager.map_info[id_count] = info
+                    address_manager.map_addr[info.peer_info.host] = id_count
+                    address_manager.tried_matrix[tried_bucket][tried_bucket_pos] = id_count
+                    address_manager.id_count += 1
+                    address_manager.tried_count += 1
+
+        # process
+        for node_id, bucket in new_table_nodes:
+            if node_id >= 0 and node_id < address_manager.new_count:
+                info = address_manager.map_info[node_id]
+                bucket_pos = info.get_bucket_position(address_manager.key, True, bucket)
+                if address_manager.new_matrix[bucket][bucket_pos] == -1 and info.ref_count < NEW_BUCKETS_PER_ADDRESS:
+                    info.ref_count += 1
+                    address_manager.new_matrix[bucket][bucket_pos] = node_id
+
+        # remove deads
+        for id, info in list(address_manager.map_info.items()):
+            if not info.is_tried and info.ref_count == 0:
+                address_manager.delete_new_entry_(id)
```

Comment:
this is duplicated from elsewhere.  let's replace both cases of it with a method.

Reply:
Created a new function in address_manager for this